editor.planx.uk/src/pages/FlowEditor/lib/analyticsProvider.tsx

```diff
@@ -122,12 +122,9 @@ export const AnalyticsProvider: React.FC<{ children: React.ReactNode }> = ({
 
   // Track component transition
   useEffect(() => {
-    if (shouldTrackAnalytics && analyticsId) {
-      const curLength = Object.keys(breadcrumbs).length;
-      const prevLength = Object.keys(previousBreadcrumbs).length;
-
-      if (curLength > prevLength) track("forwards", analyticsId);
-      if (curLength < prevLength) track("backwards", analyticsId);
+    if (shouldTrackAnalytics && analyticsId && node?.id) {
+      const logDirection = detemineLogDirection();
+      if (logDirection) track(logDirection, analyticsId, node.id);
```

Comment:
That's a really good question, I think the `useEffect` is triggered on any `breadcrumbs` change.

So I think this could lead to situation where the `breadcrumbs` change but the relative lengths of `breadcrumbs` and `previousBreadcrumbs` stay the same? I'm not sure when this would happen but I'm not particularly familiar of how some of this stuff works 😅 

I'm also maybe being overly defensive as I think there could be situations where the lifecycle of updating `previousBreadcrumbs` goes a bit wrong and has unanticipated behaviour. 

I think in the branch where we were updating some of this stuff we changed tack as we were finding `setPreviousBreadcrumbs` unreliable.